# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/src/bindings.rs
# Commit: 971767c6ed2b
# Full Hash: 971767c6ed2b2201e172105046ecc61942e03a7e
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-24 21:46:30
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 436df7dc06c6 -r 971767c6ed2b gfx/webrender_bindings/src/bindings.rs
--- a/gfx/webrender_bindings/src/bindings.rs	Fri Jan 24 01:04:09 2020 +0000
+++ b/gfx/webrender_bindings/src/bindings.rs	Thu Jan 23 23:52:05 2020 +0000
@@ -1220,6 +1220,10 @@
         clip_rect: DeviceIntRect,
     );
     fn wr_compositor_end_frame(compositor: *mut c_void);
+    fn wr_compositor_enable_native_compositor(
+        compositor: *mut c_void,
+        enable: bool,
+    );
 }
 
 pub struct WrCompositor(*mut c_void);
@@ -1345,6 +1349,15 @@
             );
         }
     }
+
+    fn enable_native_compositor(&mut self, enable: bool) {
+        unsafe {
+            wr_compositor_enable_native_compositor(
+                self.0,
+                enable,
+            );
+        }
+    }
 }
 
 
@@ -1585,6 +1598,11 @@
     dh.api.send_debug_cmd(DebugCommand::ClearCaches(ClearCache::all()));
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn wr_api_enable_native_compositor(dh: &mut DocumentHandle, enable: bool) {
+    dh.api.send_debug_cmd(DebugCommand::EnableNativeCompositor(enable));
+}
+
 fn make_transaction(do_async: bool) -> Transaction {
     let mut transaction = Transaction::new();
     // Ensure that we either use async scene building or not based on the